src/progress.rs

```diff
@@ -0,0 +1,56 @@
+use std::io::Read;
+
+use indicatif::{ProgressBar, ProgressStyle};
+use reqwest::blocking::Response;
+
+pub struct ResponseProgress {
+    progress: Option<ProgressBar>,
+    response: Response,
+}
+
+fn make_progress_bar(size: u64) -> ProgressBar {
+    let bar = ProgressBar::new(size);
+
+    bar.set_style(
+        ProgressStyle::with_template(
+            "[{elapsed_precise}] [{bar:40}] {bytes}/{total_bytes} ({bytes_per_sec}, {eta})",
+        )
+        .unwrap()
+        .progress_chars("#>-"),
+    );
+
+    bar
+}
+
+impl ResponseProgress {
```

Comment:
you think we should unit test this somehow?

Reply:
I added a test to ensure we still read from the underlying response with the progress wrapper. Also added a `indicatif::TermLike` impl that captures the progress bar activity and saves all the activity into a String, then checks for "#" characters in the string after reading all the data.

The mock impl is a bit verbose so I added it in a separate commit to revert it easily. Let me know if you'd prefer to just test the read operations pass through.